src/stan/mcmc/hmc/hamiltonians/base_hamiltonian.hpp

```diff
@@ -36,6 +36,8 @@ namespace stan {
         return T(z) + V(z);
       }
 
+      virtual double dG_dt(Point& z) = 0;
```

Comment:
Can you add some doc at least in the base class explaining what dG_dt is?  Just a note and maybe a pointer to a paper.